### PR TITLE
Docstring updates for upcoming doc publish

### PR DESCRIPTION
**Pull request by mattdangerw**

### CODE REVIEW

File: keras_nlp/layers/fnet_encoder.py [hunk 1]
```diff
@@ -21,12 +21,11 @@
...
-    "FNet: Mixing Tokens with Fourier Transforms"
-    (https://arxiv.org/abs/2105.03824). Users can instantiate multiple instances
-    of this class to stack up the encoder.
+    This class follows the architecture of FNet encoder layer in the
+    [FNet paper](https://arxiv.org/abs/2105.03824). Users can instantiate
+    multiple instances of this class to stack up the encoder.
 
-    Note on padding: In the official FNet code, padding tokens are added to the
+    Note on masking: In the official FNet code, padding tokens are added to the
     the input. However, the padding masks are deleted, i.e., mixing of
```
Comment (chenmoneygithub) [inline keras_nlp/layers/fnet_encoder.py:29]: Rereading this - I feel the sentence after i.e., makes it more confusing. What is "mixing of all tokens"?

File: keras_nlp/tokenizers/tokenizer.py [hunk 2]
```diff
@@ -20,32 +20,42 @@
...
+    Subclassers should always implement the `tokenize()` method, which will also
+    be the default when calling the layer directly on inputs.
 
-    Subclassers should implement the `detokenize()` method if the tokenization
-    is reversible. Otherwise, this can be skipped.
+    Subclassers can optionally implement the `detokenize()` method if the
+    tokenization is reversible. Otherwise, this can be skipped.
 
     Subclassers should implement `get_vocabulary()`, `vocabulary_size()`,
     `token_to_id()` and `id_to_token()` if applicable. For some simple
```
Comment (chenmoneygithub) [inline keras_nlp/tokenizers/tokenizer.py:35]: May just say vocab-free tokenizers?

File: keras_nlp/tokenizers/unicode_character_tokenizer.py [hunk 3]
```diff
@@ -108,7 +126,7 @@ class UnicodeCharacterTokenizer(tokenizer.Tokenizer):
         [2346, 2369, 2360, 2381, 2340],
         [1705, 1578, 1575, 1576,    0]], dtype=int32)>
 
-    Batch first, then tokenize for dense outputs (`sequence_length` provided).
+    Batch, then tokenize for dense outputs (`sequence_length` provided).
     (`sequence_length` provided).
```
Comment (chenmoneygithub) [inline keras_nlp/tokenizers/unicode_character_tokenizer.py:130]: duplicated (`sequence_length` provided)

File: keras_nlp/tokenizers/word_piece_tokenizer.py [hunk 4]
```diff
@@ -83,12 +83,23 @@ class WordPieceTokenizer(tokenizer.Tokenizer):
     `lowercase=False`, `strip_accents=False` and `split=False`. In
     this case, inputs should be pre-split string tensors or ragged tensors.
 
+    Tokenizer outputs can either be padded and truncated with a
+    `sequence_length` argument, or left un-truncated. The exact output will
```
Comment (chenmoneygithub) [inline keras_nlp/tokenizers/word_piece_tokenizer.py:87]: Not a blocking comment - this part appears to be duplicated in all tokenizer subclasses, do we want to move it into the base tokenizer?

**Review by chenmoneygithub — CHANGES_REQUESTED** [review]
thanks! mainly lgtm

File: keras_nlp/tokenizers/word_piece_tokenizer.py (hunk 4 above)
Comment (mattdangerw) [reply keras_nlp/tokenizers/word_piece_tokenizer.py:87]: I'm worried about discover-ability. If we only document this on the base class, no one will find it. We this to show up in the documentation for every tokenizer.

Also not a contract, it's just a convention.

File: keras_nlp/layers/fnet_encoder.py (hunk 1 above)
Comment (mattdangerw) [reply keras_nlp/layers/fnet_encoder.py:29]: I think a reference to the paper basically. You can view attention as a mechanism for "mixing tokens" together and a fourier as an alternative mixing approach.

Anyway, I just wanted to clarify that this is relevant to the usual masking behavior of an encoding block.

File: keras_nlp/tokenizers/word_piece_tokenizer.py (hunk 4 above)
Comment (chenmoneygithub) [reply keras_nlp/tokenizers/word_piece_tokenizer.py:87]: We can let this PR go for now, but we can possibly do similar treatment like: https://github.com/keras-team/keras/blob/master/keras/optimizers/optimizer_experimental/adadelta.py#L140

we can do it as a followup

**Review by chenmoneygithub — APPROVED** [review]

File: keras_nlp/tokenizers/word_piece_tokenizer.py (hunk 4 above)
Comment (mattdangerw) [reply keras_nlp/tokenizers/word_piece_tokenizer.py:87]: Ah yeah makes sense. Let's consider that as a follow up.

I don't think it is fair to demand all tokenizers support `sequence_length`, in fact, the example we give on the base class does not. It's more a convention.

Comment (mattdangerw) [reply keras_nlp/tokenizers/word_piece_tokenizer.py:87]: Though maybe we could consider a redesign here where that logic was contained on the base class. Will keep thinking on that. Anyway a follow up :)

**Pull request merged**